openexr: add 2.4.3, 2.5.9, 3.1.11 & 3.2.0

aumuell opened the pull request:
- 2.5.9 is the latest version compatible with OpenSceneGraph
- improved compatibility with GCC 13

This includes review suggestions from and supersedes #39163, but also drops adding outdated patch releases, especially 2.5.9, as it is 2.5.8 + the patch for GCC 13 compatibility.

aumuell committed:
openexr: add 2.4.3, 2.5.9, 3.1.11 & 3.2.0

- 2.5.9 is the latest version compatible with OpenScenGraph
- improved compatibility with GCC 13

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

tldahlgren reviewed (commented):
Confirmed the four new version sha256.

adamjstewart commented:
Same comments as last PR. I'm not maintaining this package so feel free to ignore my comments.

aumuell committed:
openexr: transform into multi-build system package

simplifies package considerably, as nothing special seems to be required

aumuell committed:
openexr: drop unsupported debug variant

--disable/enable-debug is not supported by ./configure in 1.3, 1.7, 2.0, 2.2 and 2.3

aumuell committed:
openexr: pkg-config is also used by @3

aumuell committed:
openexr: use system libdeflate instead of internal

if no libdeflate is found, openexr would download and build its own starting with 3.2.0

aumuell committed:
openexr: disable tests

would download lots of data during cmake and make build times noticably longer

aumuell commented:
@adamjstewart I converted it to a multi-build package, which made things simpler.

While doing this I discovered some issues, which I fixed

adamjstewart commented:
@spackbot run pipeline
